packages/mwp-core/src/renderers/server-render.jsx

```diff
@@ -53,9 +54,34 @@ const resolveSideEffects = () => ({
 	head: Helmet.rewind(),
 	redirect: Redirect.rewind(),
 	forbidden: Forbidden.rewind(),
-	notFound: NotFound.rewind(),
+	notFound: NotFound.rewind()
 });
 
+/**
+ * Using the User agent string + an external library called Mobile-Device we detect if the user
+ * is attempting to render the page on the server using a mobile, tablet, or desktop device
+ * @param  {String} uas User agent string detected from headers
+ * @return {Object}     Object of which category the devices falls within
+ */
+const getDeviceType = uas => {
+	// we may be re-writing the uas in fastly for caching purposes, so check that first
+	if (uas === 'mobile' || uas === 'desktop') {
+		return {
+			isMobile: uas === 'mobile',
+			isDesktop: uas === 'desktop',
+		}
+	}
+	// parses user agent string to determine if user is on mobile / tablet / desktop device
+	const md = new MobileDetect(uas);
+	const isMobile = md.mobile() !== null && md.phone() !== null;
+	const isTablet = md.tablet() !== null;
+	return {
+		isMobile,
+		isTablet,
+		isDesktop: !isMobile && !isTablet
```

Comment:
I'm not sure we need to have a map of `is<Device>` keys that exactly parallels the `isAt<Size>Up` map defined by `withMatchMedia` - why not use the same map in `state.config` that will be applied by `connectedWithMatchMedia`? It keeps the 'data types' much simpler and reduces the indirection.

Reply:
Yeah that's definitely a better way to go.